camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelSimpleAnnotator.java

```diff
@@ -69,9 +69,14 @@ void validateText(@NotNull PsiElement element, @NotNull AnnotationHolder holder,
                     if (!result.isSuccess()) {
                         String error = result.getShortError();
                         TextRange range = element.getTextRange();
+                        int startIdx = result.getIndex() == 0 ? text.indexOf("$") : result.getIndex();
+
+                        int endIdx = text.indexOf("}", startIdx);
+                        endIdx = endIdx == -1 ? text.indexOf(" ", startIdx) - 1 : endIdx;
+
+                        int propertyLength = (((endIdx < 0 ? text.length() - 1 : endIdx) + 1) - startIdx) + 1;
                         if (result.getIndex() > 0) {
-                            // use -1 to skip the last quote sign
-                            range = TextRange.create(range.getStartOffset() + result.getIndex(), range.getEndOffset() - 1);
+                            range = TextRange.create(range.getStartOffset() + startIdx + 1, range.getStartOffset() + startIdx + propertyLength);
```

Comment:
We can implement more safeguards and more comment. But the previous solution had the limit that it always assumed it was to the end of the element and to much was highlighted

With the new solution I try to solve 2 issue : 
1. Find the first " " or } after the getIndex
2. if the expression is never ended with " " or "}" we should use the range.getEndOffset()

I will try to construct in another way so it's more readable